### ISSUE TITLE
Adding validation loss computation and logging during fine-tuning

### ISSUE THREAD

**Issue by jjottar** [Physical-Intelligence/openpi#544]
Hi and thank you for the great work!

# Context

I recently modified the training script to allow for an optional validation dataset/repo which is used to compute and log the validation loss at regular intervals during the training loop. While I understand the most relevant validation is the one done after deploying the policy to perform the tasks in a real/sim environment, I find that having the validation loss logged during training is helpful to e.g. know when to stop fine-tuning.

# Proposal

You can find the changes in this [branch](https://github.com/jjottar/openpi-aml/tree/feature/validation_loss), in particular in this [commit](https://github.com/jjottar/openpi-aml/commit/647768301ff5ba1b1577f8ec5a12ca786cd78555).

In a nutshell, the changes are:

1. Add the validation-related config variables in src/openpi/training/config.py.
2. Modify the scripts/train.py script to compute and log the validation loss at regular intervals during the training loop.

If you’d welcome the change, I’d be happy to open a PR (we can of course discuss the details of the implementation if the current one is not satisfactory). 

Some features of the current implementation:

- The validation loss logging frequency, as well as the number and size of the validation batches to be used in the calculation, are configurable.
- For consistency, the norm_stats from the training data are passed to the validation loss computation.
- The validation dataset/repo is first searched for locally, and if not found a download is attempted from HF. If not available/otherwise fail, the validation loss computation is gracefully skipped and the training job continues as usual.

The fine-tuning job can now be run with the extra --val-repo-id flag like this (in the example below —val-repo-id is pointing to the same training repo for testing purposes — naturally, in practice it would be pointing to an unseen 10% of the data or so):

`XLA_PYTHON_CLIENT_MEM_FRACTION=0.9 uv run scripts/train.py pi0_fast_libero --exp-name=“finetune_with_val_loss” --overwrite --val-repo-id "physical-intelligence/libero"`

**Note:** in case it’s of interest, for my own testing purposes I’ve also modified examples/libero/convert_libero_data_to_lerobot.py and scripts/compute_norm_stats.py to A) generate a train/val split (e.g. 90%-10%) for the LIBERO dataset starting from the raw files and B) compute the norm_stats on the 90% used for training. I could also PR these changes if you’re interested. However, as you're aware the convert_libero_data_to_lerobot.py does no longer run with the current environment due to the refactoring of LeRobot that took place a few weeks back, so for that script to run the repo needs to be rolled back to a specific commit before those changes. 

Here's an example of the validation loss being computed and logged concurrently with the training loss during a fine-tuning run of pi0-fast, for a random 90%-10% train-val split of the episodes in physical-intelligence/libero:

![Image](https://github.com/user-attachments/assets/51ce2d0f-0e62-43f9-9210-84e09a544971)

**Comment by jjottar** [comment]
@uzhilinsky would the proposed contribution be of interest for the team?
